thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/validators/DetectionConfigValidator.java

```diff
@@ -104,80 +112,155 @@ public void validateConfig(DetectionConfigDTO detectionConfig) throws IllegalArg
-    Preconditions.checkArgument(detectionYaml.containsKey(PROP_METRIC), "Property missing " + PROP_METRIC);
-    Preconditions.checkArgument(detectionYaml.containsKey(PROP_DATASET), "Property missing " + PROP_DATASET);
-    Preconditions.checkArgument(detectionYaml.containsKey(PROP_RULES), "Property missing " + PROP_RULES);
+    Preconditions.checkArgument(detectionYaml.containsKey(PROP_METRIC),
+        "Missing property (" + PROP_METRIC + ") in sub-alert " + alertName);
+    Preconditions.checkArgument(detectionYaml.containsKey(PROP_DATASET),
+        "Missing property (" + PROP_DATASET + ") in sub-alert " + alertName);
+    Preconditions.checkArgument(detectionYaml.containsKey(PROP_RULES),
+        "Missing property (" + PROP_RULES + ") in sub-alert " + alertName);
 
-    // Validate fields which shouldn't be defined at root level
-    Preconditions.checkArgument(!detectionYaml.containsKey(PROP_FILTER), "Please double check the filter"
-        + " config. Adding dimensions filters should be in the yaml root level using 'filters' as the key. Anomaly "
-        + "filter should be added in to the indentation level of detection yaml it applies to.");
+    // Validate fields which shouldn't be defined at this level
+    Preconditions.checkArgument(!detectionYaml.containsKey(PROP_FILTER),
+        "For sub-alert " + alertName + ", please double check the filter config. Adding dimensions filters"
+            + " should be in the yaml root level using 'filters' as the key. Anomaly filter should be added in to the"
+            + " indentation level of detection yaml it applies to.");
 
     // Check if the metric defined in the config exists
     MetricConfigDTO metricConfig = provider
         .fetchMetric(MapUtils.getString(detectionYaml, PROP_METRIC), MapUtils.getString(detectionYaml, PROP_DATASET));
-    Preconditions.checkNotNull(metricConfig, "Metric defined in the config cannot be found");
-    DatasetConfigDTO datasetConfig = provider
-        .fetchDatasets(Collections.singletonList(metricConfig.getDataset()))
-        .get(metricConfig.getDataset());
+    Preconditions.checkArgument(metricConfig != null,
+        "Invalid metric (not found) in sub-alert " + alertName);
 
     // We support only one grouper per metric
     Preconditions.checkArgument(ConfigUtils.getList(detectionYaml.get(PROP_GROUPER)).size() <= 1,
-        "Multiple groupers detected for metric. We support only one grouper per metric.");
+        "Multiple groupers detected for metric in sub-alert " + alertName);
 
     // Validate all the rules
     Set<String> names = new HashSet<>();
     List<Map<String, Object>> ruleYamls = ConfigUtils.getList(detectionYaml.get(PROP_RULES));
-    for (int i = 1; i <= ruleYamls.size(); i++) {
-      Map<String, Object> ruleYaml = ruleYamls.get(i - 1);
+    for (int ruleIndex = 1; ruleIndex <= ruleYamls.size(); ruleIndex++) {
```

Comment:
Why the index starts from 1?

Reply:
It doesn't really affect anything. I just retained the existing convention.

Reply:
Not intuitive. Suggest to change it. Although it is small..